document/document_test.go

```diff
@@ -656,12 +665,4 @@ func BenchmarkDocumentIterate(b *testing.B) {
 		}
 	})
 
-	b.Run("Reflection", func(b *testing.B) {
-		for i := 0; i < b.N; i++ {
-			refd, _ := document.NewFromStruct(&f)
-			refd.Iterate(func(string, document.Value) error {
-				return nil
-			})
-		}
-	})
```

Comment:
Could you explain why this benchmark was removed?

Reply:
It because now we are using a `FieldBuffer`, @asdine told me to delete it for this reason.

> * Perhaps we should be rename `NewFromStruct` to `NewFieldBufferFromStruct` since it’s now building the `FieldBuffer`?
 `NewFromStruct` returns a `Document`. 


> * Why can’t we keep the `structDocument` and descend into nested structs / anonymous fields on iteration?
I think using `FieldBuffer` is a good way because it already implements `GetByField` and `Iterate`.

Reply:
> It because now we are using a FieldBuffer

Ah, that’s right.

>I think using FieldBuffer is a good way because it already implements GetByField and Iterate.

I’m fine with both approaches, though I definitely have a preference for the laziness of `structDocument`.

Reply:
I have the same preference, but since `NewFromStruct` is mostly used when inserting data, I don't have strong reasons to refuse this change.